File: src/test/java/com/otus/mock/QuizServiceTestConfig.java
package com.otus.mock;

import com.otus.repository.CsvRepository;
import com.otus.service.QuizService;
import com.otus.service.QuizServiceImpl;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

import static org.mockito.Mockito.mock;

@Configuration
public class QuizServiceTestConfig {

    @Bean
    public QuizService quizService(){
        return new QuizServiceImpl(csvRepository(), null);
    }

    @Bean
    @Primary
    public CsvRepository csvRepository() {
        return mock(CsvRepository.class);
    }
}

File: src/test/java/com/otus/mock/QuizServiceTestConfig.kt
package com.otus.mock

import com.otus.repository.CsvRepository
import com.otus.service.AnswerService
import com.otus.service.AnswerServiceImpl
import com.otus.service.QuizServiceImpl
import org.mockito.Mockito.mock
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class QuizServiceTestConfig {

    @Bean
    open fun quizService() = QuizServiceImpl(csvRepository(), null)

    @Bean
    open fun csvRepository() = mock(CsvRepository::class.java)!!
}